sdk/search/Azure.Search.Documents/src/Batching/PublisherAction{T}.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using Azure.Search.Documents.Models;
+
+namespace Azure.Search.Documents.Batching
+{
+    /// <summary>
+    /// Tracks actions that the publisher hasn't submitted yet.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The .NET type that maps to the index schema.  Instances of this
+    /// type can be retrieved as documents from the index.
+    /// </typeparam>
+    internal class PublisherAction<T>
```

Comment:
super nit: You might be able to use a mutable struct for this.

Reply:
I gave that a bit of thought and got worried that once we added pre-serialized JSON bytes it wouldn't make as much sense.  It's all internal though so I could flip it for now and change back later.